client/rest-high-level/src/main/java/org/elasticsearch/client/dataframe/transforms/DataFrameTransformConfig.java

```diff
@@ -0,0 +1,170 @@
+import org.elasticsearch.index.query.MatchAllQueryBuilder;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import static org.elasticsearch.common.xcontent.ConstructingObjectParser.constructorArg;
+import static org.elasticsearch.common.xcontent.ConstructingObjectParser.optionalConstructorArg;
+
+public class DataFrameTransformConfig implements ToXContentObject {
+
+    public static final ParseField ID = new ParseField("id");
+    public static final ParseField SOURCE = new ParseField("source");
+    public static final ParseField DEST = new ParseField("dest");
+    public static final ParseField QUERY = new ParseField("query");
+    // types of transforms
+    public static final ParseField PIVOT_TRANSFORM = new ParseField("pivot");
+
+    private final String id;
+    private final String source;
+    private final String dest;
+    private final QueryConfig queryConfig;
+    private final PivotConfig pivotConfig;
+
+    private static final ConstructingObjectParser<DataFrameTransformConfig, String> PARSER =
+            new ConstructingObjectParser<>("data_frame_transform", true,
+                (args) -> {
+                    String id = (String) args[0];
+                    String source = (String) args[1];
+                    String dest = (String) args[2];
+                    // default handling: if the user does not specify a query, we default to match_all
+                    QueryConfig queryConfig = (args[3] == null) ? new QueryConfig(new MatchAllQueryBuilder()) : (QueryConfig) args[3];
+                    PivotConfig pivotConfig = (PivotConfig) args[4];
+                    return new DataFrameTransformConfig(id, source, dest, queryConfig, pivotConfig);
+                });
+
+    static {
+        PARSER.declareString(optionalConstructorArg(), ID);
+        PARSER.declareString(constructorArg(), SOURCE);
+        PARSER.declareString(constructorArg(), DEST);
+        PARSER.declareObject(optionalConstructorArg(), (p, c) -> QueryConfig.fromXContent(p), QUERY);
+        PARSER.declareObject(optionalConstructorArg(), (p, c) -> PivotConfig.fromXContent(p), PIVOT_TRANSFORM);
+    }
+
+    public static DataFrameTransformConfig fromXContent(final XContentParser parser) {
+        return PARSER.apply(parser, null);
+    }
+
+
+    public DataFrameTransformConfig(final String id,
+                                    final String source,
+                                    final String dest,
+                                    final QueryConfig queryConfig,
+                                    final PivotConfig pivotConfig) {
+        this.id = Objects.requireNonNull(id);
+        this.source = Objects.requireNonNull(source);
+        this.dest = Objects.requireNonNull(dest);
+        this.queryConfig = queryConfig;
+        this.pivotConfig = pivotConfig;
+    }
+
+    public String getId() {
+        return id;
+    }
+
+    // TODO should this function be removed?
+    public String getCron() {
+        return "*";
```

Comment:
This cannot be set and not appear in the xContent representation. Should it be dropped?